src/main/java/com/okta/sdk/clients/UserApiClient.java

```diff
@@ -296,6 +593,109 @@ public LoginCredentials changeRecoveryQuestion(String userId, String currentPass
+    }
+
+    /**
+     * Get all users up to a limit using cursor with paging info.
+     *
+     * @param  after               The cursor that determines which users to return after.
+     * @param  limit               The max number of results returned.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsAfterCursorWithLimit(String after, int limit) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseAfterCursorWithLimit(after, limit));
+    }
+
+    /**
+     * Get all users from url with paging info.
+     *
+     * @param  url                 The URL to get users from.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsByUrl(String url) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseByUrl(url));
+    }
+
+    /**
+     * Get all users using advanced search with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearch(FilterBuilder filterBuilder) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseWithAdvancedSearch(filterBuilder));
+    }
+
+    /**
+     * Get all users up to a limit using advanced search with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @param  limit               The max number of results returned.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearchAndLimit(FilterBuilder filterBuilder, int limit) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseWithAdvancedSearchAndLimit(filterBuilder, limit));
+    }
+
+    /**
+     * Get all users up to a limit using advanced search and cursor with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @param  limit               The max number of results returned.
+     * @param  after               The cursor that determines which users to return after.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearchAndLimitAndAfterCursor(FilterBuilder filterBuilder, int limit, String after) throws IOException {
```

Comment:
Looking at this again and thinking through it more, I like it the way you have it. Thanks for the explanation as well.